tempo_embeddings/embeddings/model.py

```diff
@@ -114,14 +114,24 @@ def layer(self, layer: int) -> None:
     def name(self) -> str:
         return self._model.config._name_or_path  # pylint: disable=protected-access
 
-    def _batches(self, passages: list[Passage]) -> Iterable[list[Passage]]:
-        for batch_start in tqdm(
-            range(0, len(passages), self.batch_size),
-            desc="Embeddings",
-            unit="batch",
-            total=len(passages) // self.batch_size + 1,
-        ):
-            yield passages[batch_start : batch_start + self.batch_size]
+    def _batches(self, passages: list[Passage], skip_batching: bool = False) -> Iterable[list[Passage]]:
+        if skip_batching:
+            yield passages
+        else:
+            for batch_start in tqdm(
+                range(0, len(passages), self.batch_size),
+                desc="Embeddings",
+                unit="batch",
+                total=len(passages) // self.batch_size + 1,
+            ):
+                yield passages[batch_start : batch_start + self.batch_size]
+
+    def _get_token_spans(self, encoding: Encoding) -> list[tuple[int, int]]:
```

Comment:
This is a pure function which does not use `self`, so it can be static.

```suggestion
    @staticmethod
    def _get_token_spans(encoding: Encoding) -> list[tuple[int, int]]:
```
I think this has been introduced earlier, but now might be a good occasion to fix it.